src/dataregistry/query.py

```diff
@@ -448,3 +449,140 @@ def get_dataset_absolute_path(self, dataset_id, schema=None):
         else:
             print(f"No dataset with dataset_id={dataset_id}")
             return None
+
+    def resolve_alias(self, alias):
+        """
+        Parameters
+        ----------
+        alias      String or int      Either name or id of an alias
+
+        Returns
+        -------
+        id         int                id of item (dataset or alias)
+                                      referred to
+        ref_type   string             type of object aliased to,
+                                      either "dataset" or "alias"
+        """
+        tbl = self._tables["dataset_alias"]
+        if isinstance(alias, int):
+            filter_column = "dataset_alias.dataset_alias_id"
+        elif isinstance(alias, str):
+            filter_column = "dataset_alias.alias"
+        else:
+            raise ValueError("Argument 'alias' must be int or str")
+        f = Filter(filter_column, "==", alias)
+
+        stmt = select(tbl.c.dataset_id, tbl.c.ref_alias_id)
+        stmt = stmt.select_from(tbl)
+        stmt = self._render_filter(f, stmt)
+
+        with self._engine.connect() as conn:
+            try:
+                result = conn.execute(stmt)
+            except DBAPIError as e:
+                print("Original error:")
+                print(e.StatementError.orig)
+                return None
+
+        row = result.fetchone()
+        if not row:
```

Comment:
i think returning none is fine, can add to the doc string, if not found returns none

Reply:
Ok.  Updating docstring for next commit.